dlt/dlt_broker_handlers.py

```diff
@@ -306,6 +295,120 @@ def handle(self, message):
+
+    This process instance is responsible for collecting messages from
+    the at-runtime written dlt log, tagging them with the correct queue id and placing
+    them on the messages queue.
+    """
+
+    def __init__(
+        self, filter_queue, message_queue, mp_stop_event, file_name, dlt_time_value=None, filter_ack_queue=None
+    ):
+        super().__init__(filter_queue, message_queue, mp_stop_event, dlt_time_value, filter_ack_queue)
+        self.file_name = file_name
+        self.dlt_reader = load(filename=self.file_name, live_run=True)
+
+    def is_valid_message(self, message):
+        """According to AUTOSAR doc, message with empty apid and empty ctid is still valid"""
+        return message is not None
+
+    def run(self):
+        """DLTFileSpinner worker method"""
+        logger.info("Start to process dlt file %s", self.file_name)
+        # Even though dlt connector for ioc should only be instantiated after successful SerialConsole with fibex,
+        # the corner case of not-existing dlt file will still be handled here with max 5 retires
+        retries_for_non_existing_file = 5
+
+        while not self.mp_stop_flag.is_set():
+            try:
+                logger.debug("py_dlt_file_main_loop")
+                res = py_dlt_file_main_loop(self.dlt_reader, callback=self.handle)
+                if res is False and not self.mp_stop_flag.is_set():  # main loop returned False
+                    logger.error("Too many bad messages read from %s", self.file_name)
+                    self.mp_stop_flag.set()
+                    break
+            except KeyboardInterrupt:
+                logger.debug("main loop manually interrupted")
+                break
+            except IOError as err:
+                if str(err) == cDLT_FILE_NOT_OPEN_ERROR:
+                    # Not every time of non-existing file, cDLTFile will report error
+                    # Sometimes, it simply works through without issue.
+                    # So, no unittest could be done for this error handling
+                    if retries_for_non_existing_file == 0:
+                        logger.error("After retries, dlt file %s still does not exist", self.file_name)
+                        raise err
+                    logger.warning(
+                        "DLT file %s does not exist, will try %d times again",
+                        self.file_name,
+                        retries_for_non_existing_file,
+                    )
+                    retries_for_non_existing_file = retries_for_non_existing_file - 1
+                    time.sleep(1)
+                else:
+                    raise err
+            except Exception:  # pylint: disable=broad-except
+                logger.exception("Exception during the DLT message receive")
+
+        logger.debug("DLTFileSpinner starts to quit...")
+        if self.dlt_reader.stop_reading_proc.is_set():
+            self.dlt_reader.stop_reading_proc.set()
```

Comment:
aaa what? why are you checking if it's set before setting it? :open_mouth:

Reply:
emm, good catch. a "not" is missing...